src/pages/common/_voiceconfig_page.c

```diff
@@ -122,26 +122,36 @@ static const char *voicelbl_cb(guiObject_t *obj, const void *data)
     int idx = (long)data;
     switch (voiceconfig_getsrctype(idx)) {
         case VOICE_SRC_SWITCH:
-            if (Model.voice.switches[idx - VOICE_SRC_SWITCH].music)
-                return voice_map[Model.voice.switches[idx - VOICE_SRC_SWITCH].music].label;
+            if (Model.voice.switches[idx - VOICE_SRC_SWITCH].music) {
+                CONFIG_VoiceParse(voice_map[Model.voice.switches[idx - VOICE_SRC_SWITCH].music].id);
+                return tempstring;
+            }
             break;
 #if NUM_AUX_KNOBS
         case VOICE_SRC_AUX:
-            if(Model.voice.aux[idx - VOICE_SRC_AUX].music)
-                return voice_map[Model.voice.aux[idx - VOICE_SRC_AUX].music].label;
+            if(Model.voice.aux[idx - VOICE_SRC_AUX].music) {
+                CONFIG_VoiceParse(voice_map[Model.voice.aux[idx - VOICE_SRC_AUX].music].id);
+                return tempstring;
+            }
```

Comment:
I really don't like this.  A function should not use tempstring as an implicit argument/return passing  If you need to pass it, have CONFIG_VoiceParse return tempstring, and here just run retun CONFIG_VoiceParse(...)

Reply:
I hope the solution is better now. It is definitely smaller ;)